dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java

```diff
@@ -27,23 +49,178 @@ public ClusterEndpoint(final TopologyManagementRequestSender requestSender) {
+      case final MemberJoinOperation join -> new Operation()
+          .operation(OperationEnum.BROKER_ADD)
+          .brokerId(Integer.parseInt(join.memberId().id()));
+      case final MemberLeaveOperation leave -> new Operation()
+          .operation(OperationEnum.BROKER_REMOVE)
+          .brokerId(Integer.parseInt(leave.memberId().id()));
+      case final PartitionJoinOperation join -> new Operation()
+          .operation(OperationEnum.PARTITION_JOIN)
+          .brokerId(Integer.parseInt(join.memberId().id()))
+          .partitionId(join.partitionId())
+          .priority(join.priority());
+      case final PartitionLeaveOperation leave -> new Operation()
+          .operation(OperationEnum.PARTITION_LEAVE)
+          .brokerId(Integer.parseInt(leave.memberId().id()))
+          .partitionId(leave.partitionId());
+      case final PartitionReconfigurePriorityOperation reconfigure -> new Operation()
+          .operation(OperationEnum.PARTITION_RECONFIGURE_PRIORITY)
+          .brokerId(Integer.parseInt(reconfigure.memberId().id()))
+          .partitionId(reconfigure.partitionId())
+          .priority(reconfigure.priority());
+    };
+  }
+
+  private static List<BrokerState> mapBrokerStates(final Map<MemberId, MemberState> topology) {
+    return topology.entrySet().stream()
+        .map(
+            entry ->
+                new BrokerState()
+                    .id(Integer.parseInt(entry.getKey().id()))
+                    .state(mapBrokerState(entry.getValue().state()))
+                    .lastUpdatedAt(entry.getValue().lastUpdated().atOffset(ZoneOffset.UTC))
+                    .version(entry.getValue().version())
+                    .partitions(mapPartitionStates(entry.getValue().partitions())))
+        .toList();
+  }
+
+  private static BrokerStateCode mapBrokerState(final MemberState.State state) {
+    return switch (state) {
+      case ACTIVE -> BrokerStateCode.ACTIVE;
+      case JOINING -> BrokerStateCode.JOINING;
+      case LEAVING -> BrokerStateCode.LEAVING;
+      case LEFT -> BrokerStateCode.LEFT;
+      case UNINITIALIZED -> BrokerStateCode.UNKNOWN;
+    };
+  }
+
+  private static List<PartitionState> mapPartitionStates(
+      final Map<Integer, io.camunda.zeebe.topology.state.PartitionState> partitions) {
+    return partitions.entrySet().stream()
+        .map(
+            entry ->
+                new PartitionState()
+                    .id(entry.getKey())
+                    .priority(entry.getValue().priority())
+                    .state(mapPartitionState(entry.getValue().state())))
+        .toList();
+  }
+
+  private static PartitionStateCode mapPartitionState(final State state) {
+    return switch (state) {
+      case JOINING -> PartitionStateCode.JOINING;
+      case ACTIVE -> PartitionStateCode.ACTIVE;
+      case LEAVING -> PartitionStateCode.LEAVING;
+      case UNKNOWN -> PartitionStateCode.UNKNOWN;
+    };
+  }
+
+  public enum Resource {
     BROKERS,
+    PARTITIONS,
```

Comment:
POST /actuator/cluster/something/0/partitions/2 results in a 400 bad request, apparently from Spring directly because "something" can't be converted to one of the enums.

Nonsensical requests such as POST /actuator/cluster/brokers/0/brokers/2 are manually mapped to 404s.